Assets/QA/Tests/XRDeviceActions/_SharedScripts/Vector3ControlActionStatus.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+using UnityEngine.Experimental.Input;
+using UnityEngine.Experimental.Input.Controls;
+using UnityEngine.UI;
+
+public class Vector3ControlActionStatus : MonoBehaviour {
+
+	public InputAction vector3Action;
+
+    public Text statusText;
+
+	// Use this for initialization
+	void Start () {
+        vector3Action.Enable();
+        vector3Action.performed += UpdateVector3;
+        vector3Action.started += UpdateVector3;
+        vector3Action.cancelled += UpdateVector3;
+	}
+
+    private void UpdateVector3(InputAction.CallbackContext context)
+    {
+        Vector3 value = ((Vector3Control)(context.control)).ReadValue();
+        statusText.text = Vector3ToFieldText(value);
+    }
+
+    private string Vector3ToFieldText (Vector3 inVec)
```

Comment:
I would suggest using the formatting version of this [https://docs.unity3d.com/ScriptReference/Vector3.ToString.html] instead.

Reply:
I chose to do it this way, and with a monospace font, so that it is super easy to read.  Using the other method makes UI spacing difficult and can result in ugly and difficult to decipher results.

Reply:
Wouldn't this still be functionally equivalent to?
`inVec.ToString("+000.000;-000.000;+000.000");`
The Vector3 function is this:
```
public string ToString(string format)
        {
            return UnityString.Format("({0}, {1}, {2})", x.ToString(format), y.ToString(format), z.ToString(format));
        }
```

Duplicating the same behaviour in two places is not a good idea.